src/printer.js

```diff
@@ -5336,11 +5357,36 @@ function hasDanglingComments(node, filter) {
   )
 }
 
-function willPrintOwnComments(path) {
+function isInlineLeadingComment(comment, node, options) {
+  return (
+    comment.leading &&
+    !util.hasNewlineInRange(
+      options.originalText,
+      options.locEnd(comment),
+      options.locStart(node)
+    )
+  )
+}
+
+function expressionStatementHasInlineLeadingComment(node, options) {
```

Comment:
I guess this doesn't have to be `expressionStatement`-specific? It doesn't check that its arg is in fact an `ExpressionStatement`